TSPL.docc/LanguageGuide/Generics.md

```diff
@@ -1853,6 +1853,286 @@ protocol ComparableContainer: Container where Item: Comparable { }
   }
 -->
 
+## Generic Parameter Packs
+
+To understand the problem that parameter packs solve,
+consider the following overloads:
+
+```swift
+func double<T: Numeric>(_ value: T) -> T {
+    return 2 * value
+}
+
+func double<T: Numeric, U: Numeric>(_ value1: T, _ value2: U) -> (T, U) {
+    return (2 * value1, 2 * value2)
+}
+
+func double<T: Numeric, U: Numeric, V: Numeric>(
+        _ value1: T,
+        _ value2: U,
+        _ value3: V) -> (T, U, V) {
+    return (2 * value1, 2 * value2, 2 * value3)
+}
+```
+
+Each function in the example above
+takes a different number of arguments,
+and returns a tuple containing the result of doubling those arguments.
+Because the functions are generic,
+they can take values of any numeric type,
+and each argument can be a different type ---
+and the doubled values that are returned have those same types.
+Manually writing out each function like this
+is repetitive and can be error prone.
+It also imposes an arbitrary limit of three arguments,
+even though the doubling operation could really apply
+to any number of arguments.
+
+Other approaches that also have drawbacks
+include taking the arguments an array or a variadic parameter,
+which requires all arguments to be the same type,
+or using `Any` which erases type information.
+
+<!-- XXX transition -->
+
+Writing this function with a parameter pack
+preserves type information about its arguments,
+and lets you call the function with an arbitrary number of arguments:
+
+```swift
+func double<each T: Numeric>(_ value: repeat each T) -> (repeat each T) {
+    // ...
+}
+```
+
+In the code above, `Element` is a generic type parameter.
+It's marked `each Element`,
+indicating that it's a type-parameter pack.
```

Comment:
Type name inconsistency aside, I also am finding the above two sentences to be a bit confusing / potentially misleading. Starting by saying that `Element` is a generic type parameter I think is potentially misleading as it equates it to previous generic type parameters, when it is more specifically a type parameter pack. I am concerned it might make it sound like `Element` is something that it is not up until it gets "marked" with `each`, but really both are inseparable here, and we are only talking about a type parameter pack, not a singular generic type parameter.